src/AutoMapper/Mappers/FromStringDictionaryMapper.cs

```diff
@@ -33,16 +33,19 @@ public class FromStringDictionaryMapper : IObjectMapper
         private static TDestination Map<TDestination>(StringDictionary source, TDestination destination, ResolutionContext context, ProfileMap profileMap)
         {
             var destTypeDetails = profileMap.CreateTypeDetails(typeof(TDestination));
-            var members = from name in source.Keys
-                          join member in destTypeDetails.PublicWriteAccessors on name equals member.Name
-                          select member;
+
+            var memberMatches = from member in destTypeDetails.PublicWriteAccessors
+                                join key in source.Keys on member.Name equals key.Trim() into matchingKeys
+                                where matchingKeys.Any()
+                                select new { member, sourceName = matchingKeys.OrderBy(n => n.Length).First() };
+
```

Comment:
This is pretty much an arbitrary choice. I'd rather throw.

Reply:
Was debating that myself, I'll post an update soon.